components/pay/PayConfig.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Divider,
+  Flex,
+  FormControl,
+  Input,
+  InputGroup,
+  InputRightElement,
+} from "@chakra-ui/react";
+import {
+  Button,
+  Card,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  Heading,
+} from "tw-components";
+
+import { ApiKey, useUpdateApiKey } from "@3rdweb-sdk/react/hooks/useApi";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  ApiKeyPayConfigValidationSchema,
+  PERCENTAGE_TO_BPS,
+  apiKeyPayConfigValidationSchema,
+} from "components/settings/ApiKeys/validations";
+import { useTrack } from "hooks/analytics/useTrack";
+import { useTxNotifications } from "hooks/useTxNotifications";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+
+interface PayConfigProps {
+  apiKey: ApiKey;
+}
+
+const TRACKING_CATEGORY = "pay";
+
+export const PayConfig: React.FC<PayConfigProps> = ({ apiKey }) => {
+  const payService = apiKey.services?.find((service) => service.name === "pay");
+
+  const form = useForm<ApiKeyPayConfigValidationSchema>({
+    resolver: zodResolver(apiKeyPayConfigValidationSchema),
+    defaultValues: {
+      developerFeeBPS:
+        typeof payService?.developerFeeBPS === "number"
+          ? (payService?.developerFeeBPS ?? 0.0) / PERCENTAGE_TO_BPS
+          : undefined,
+      payoutAddress: payService?.payoutAddress,
+    },
+  });
+
+  useEffect(() => {
```

Comment:
we should never do form.reset and useEffect for a form, can you check how we do it in other places with transformedQueryData and setting defaultValues in the form?

Reply:
avoid useEffect as much as possible and let libraries like react-hook-form or react-query do it, useEffect triggers twice in StrictMode and can lead to issues